site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/controller/SiteController.java

```diff
@@ -32,14 +32,20 @@ public SiteController(SiteService siteService) {
 
     @PreAuthorize("@authorisationService.authorise('create-site', #site.parentSiteId)")
     @PostMapping()
-    public ResponseEntity<Response> site(@RequestBody Site site) {
+    public ResponseEntity<Response> createSite(@RequestBody Site site) {
         String siteId = siteService.save(site);
         return ResponseEntity.status(HttpStatus.CREATED).body(new Response(siteId));
     }
 
+    @PreAuthorize("@authorisationService.authorise('view-site')")
     @PostAuthorize("@authorisationService.filterUserSites(returnObject.getBody(), #role)")
     @GetMapping
-    public ResponseEntity<List<Site>> sites(@RequestParam(value = "role", required = false) String role) {
+    public ResponseEntity<List<Site>> getSites(@RequestParam(value = "role", required = false) String role) {
+        return ResponseEntity.status(HttpStatus.OK).body(siteService.findSites());
+    }
+
+    @GetMapping("/authorize")
```

Comment:
So this new GET /authorize endpoint does the same thing as the GET /sites endpoint (i.e. return all sites) but bypassing authorization?    Is this the right RESTful name for it - it's more un-authorized than authorize?   Also I wondered whether there is any alternative solution for the circularity problem in the authz component, other than a duplicate endpoint?  @LizaShak talked about this problem before but I can't remember what the conclusion was.

Reply:
In the endpoints called by the authorisation we can't call authorise. That will cause a loop. If you need an endpoint that requires a permission check... You add it on the endpoints used by authorisation.. So I guess it has to be a separate enpoint right?

Reply:
So this will be the case with all 3 endpoints called by the authz logic i.e. /sites for the site tree,  /practitioners/roles for role assignments and /roles?ids=1,2,3 for role-permission mappings?   All will need a duplicate endpoint bypassing the normal authz that would be applied to other callers?    How do we secure them?

Reply:
ok unauthorized as teh sites at that point arent authorized - I was thinking that I just wanted to make it clear that only the auth service should access the endpoint but happy to change - I think we'll end up refactoring anyhow

Reply:
Unless we can secure the duplicate endpoints, this solution is storing up a new problem?     I see that /practitioner/roles returns only the user's own roles which limits access.   But that won't work with the sites endpoint which needs to return the whole tree for the authz logic?   On the naming, /unauthorized seems confusing too now I look at it - I can't work out what the endpoint should be called and that's maybe a sign that something is up with this solution?    Would it be heresy for the authz component to get its data in another way?

Reply:
changed endpoint to sites/assigned and added the postAuth to filter which now at least fits with what the endpoint does i.e return sites after post filtering on roleassignments.